common/scala/src/main/scala/whisk/core/entity/InstanceId.scala

```diff
@@ -21,7 +21,7 @@ import spray.json.DefaultJsonProtocol
 import whisk.core.entity.ControllerInstanceId.LEGAL_CHARS
 import whisk.core.entity.ControllerInstanceId.MAX_NAME_LENGTH
 
-case class InvokerInstanceId(val instance: Int, name: Option[String] = None) {
+case class InvokerInstanceId(val instance: Int, name: Option[String] = None, hostname: Option[String] = None) {
   def toInt: Int = instance
```

Comment:
Yes please

Reply:
@tysonnorris do you use unique invoker container id as the 'name' together with dynamic kafka topic assignment?

I hesitate to use the 'name' here because of the dynamic Kafka topic assignment and its edge cases. For example in case of kube image reload, an invoker will get a new instance id which will cause a new topic creation in Kafka. This could potentially lead to some left over activations on the previous topic.